vllm/config.py

```diff
@@ -1740,7 +1740,7 @@ def _get_and_verify_max_len(
                     "with rope_scaling. Please raise an issue so we can "
                     "investigate.")
 
-            if rope_type == "mrope":
+            if rope_type in ("mrope", "default"):
```

Comment:
"mrope" gets renamed to "default" in the Qwen2-VL config class

Reply:
Qwen2-VL cannot be run in `transformers>=4.45` even with this change.

```
[rank0]: Traceback (most recent call last):
[rank0]:   File "/home/cyrusleung/vllm/examples/offline_inference_vision_language_multi_image.py", line 286, in <module>
[rank0]:     main(args)
[rank0]:   File "/home/cyrusleung/vllm/examples/offline_inference_vision_language_multi_image.py", line 262, in main
[rank0]:     run_generate(model, QUESTION, IMAGE_URLS)
[rank0]:   File "/home/cyrusleung/vllm/examples/offline_inference_vision_language_multi_image.py", line 205, in run_generate
[rank0]:     req_data = model_example_map[model](question, image_urls)
[rank0]:   File "/home/cyrusleung/vllm/examples/offline_inference_vision_language_multi_image.py", line 151, in load_qwen2_vl
[rank0]:     llm = LLM(
[rank0]:   File "/home/cyrusleung/vllm/vllm/entrypoints/llm.py", line 214, in __init__
[rank0]:     self.llm_engine = LLMEngine.from_engine_args(
[rank0]:   File "/home/cyrusleung/vllm/vllm/engine/llm_engine.py", line 564, in from_engine_args
[rank0]:     engine = cls(
[rank0]:   File "/home/cyrusleung/vllm/vllm/engine/llm_engine.py", line 325, in __init__
[rank0]:     self.model_executor = executor_class(
[rank0]:   File "/home/cyrusleung/vllm/vllm/executor/executor_base.py", line 47, in __init__
[rank0]:     self._init_executor()
[rank0]:   File "/home/cyrusleung/vllm/vllm/executor/gpu_executor.py", line 40, in _init_executor
[rank0]:     self.driver_worker.load_model()
[rank0]:   File "/home/cyrusleung/vllm/vllm/worker/worker.py", line 183, in load_model
[rank0]:     self.model_runner.load_model()
[rank0]:   File "/home/cyrusleung/vllm/vllm/worker/model_runner.py", line 1016, in load_model
[rank0]:     self.model = get_model(model_config=self.model_config,
[rank0]:   File "/home/cyrusleung/vllm/vllm/model_executor/model_loader/__init__.py", line 19, in get_model
[rank0]:     return loader.load_model(model_config=model_config,
[rank0]:   File "/home/cyrusleung/vllm/vllm/model_executor/model_loader/loader.py", line 399, in load_model
[rank0]:     model = _initialize_model(model_config, self.load_config,
[rank0]:   File "/home/cyrusleung/vllm/vllm/model_executor/model_loader/loader.py", line 176, in _initialize_model
[rank0]:     return build_model(
[rank0]:   File "/home/cyrusleung/vllm/vllm/model_executor/model_loader/loader.py", line 161, in build_model
[rank0]:     return model_class(config=hf_config,
[rank0]:   File "/home/cyrusleung/vllm/vllm/model_executor/models/qwen2_vl.py", line 876, in __init__
[rank0]:     self.model = Qwen2Model(config, cache_config, quant_config)
[rank0]:   File "/home/cyrusleung/vllm/vllm/model_executor/models/qwen2.py", line 248, in __init__
[rank0]:     self.start_layer, self.end_layer, self.layers = make_layers(
[rank0]:   File "/home/cyrusleung/vllm/vllm/model_executor/models/utils.py", line 282, in make_layers
[rank0]:     [PPMissingLayer() for _ in range(start_layer)] + [
[rank0]:   File "/home/cyrusleung/vllm/vllm/model_executor/models/utils.py", line 283, in <listcomp>
[rank0]:     maybe_offload_to_cpu(layer_fn(prefix=f"{prefix}.{idx}"))
[rank0]:   File "/home/cyrusleung/vllm/vllm/model_executor/models/qwen2.py", line 250, in <lambda>
[rank0]:     lambda prefix: Qwen2DecoderLayer(config=config,
[rank0]:   File "/home/cyrusleung/vllm/vllm/model_executor/models/qwen2.py", line 175, in __init__
[rank0]:     self.self_attn = Qwen2Attention(
[rank0]:   File "/home/cyrusleung/vllm/vllm/model_executor/models/qwen2.py", line 133, in __init__
[rank0]:     self.rotary_emb = get_rope(
[rank0]:   File "/home/cyrusleung/vllm/vllm/model_executor/layers/rotary_embedding.py", line 1003, in get_rope
[rank0]:     raise ValueError(f"Unknown RoPE scaling type {scaling_type}")
[rank0]: ValueError: Unknown RoPE scaling type default
```

If we install the older version of transformers mentioned in the docs, then vLLM cannot be run because it imports `MLlamaConfig` from the top level. We'll open a separate PR to patch in Qwen2-VL support for transformers v4.45.

Reply:
Thanks @DarkLight1337 yeah I was just making a change to update the scaling type in the config back to "mrope" if it's "default". @ywang96 found this open issue https://github.com/huggingface/transformers/issues/33401

Reply:
Change made in https://github.com/vllm-project/vllm/pull/8829/commits/ce1d477041a6808a2082e8d311ea1e5f1d4e29ac